src/test/cpp/test/AsyncLocal.cpp

```diff
@@ -0,0 +1,81 @@
+#include "catch.hpp"
+#include <windows.h>
+#include "winrt/Windows.Foundation.h"
+
+using namespace winrt;
+using namespace Windows::Foundation;
+
+namespace
+{
+    //
+    // Checks that coroutine locals are destroyed prior to notifying waiters.
+    //
+
+    struct Local
+    {
+        bool& destroyed;
+
+        ~Local()
+        {
+            Sleep(100);
+            REQUIRE(!destroyed);
+            destroyed = true;
+        }
+    };
+
+    IAsyncAction Action(HANDLE event, bool& destroyed)
+    {
+        co_await resume_on_signal(event);
+        Local local{ destroyed };
+        Sleep(100);
```

Comment:
Can you clarify why this Sleep is important?

Reply:
I'll remove those. They aren't needed. My original repro needed a sleep due to some race that prevented the repro without the sleep, but I see the simpler repro works without the sleep now.